### PR TITLE
Allow changing the default scheme for dhcpv4

### PR DESCRIPTION
**Pull request by colinmitchell-ggl**
Signed-off-by: Colin Mitchell colinmitchell@google.com

### CODE REVIEW

**Comment by colinmitchell-ggl** [comment]
The code coverage seems a little wonky, but scouts honor I did not decrease it. :)

File: pkg/dhclient/dhcp4.go [hunk 1]
```diff
@@ -172,8 +175,8 @@ func (p *Packet4) Boot() (*url.URL, error) {
 	}
 
 	if len(u.Scheme) == 0 {
-		// Defaults to tftp is not specified.
-		u.Scheme = "tftp"
+		// Use the DefaultScheme if not specified
+		u.Scheme = DefaultScheme
```
Comment (rjoleary) [inline pkg/dhclient/dhcp4.go:179]: I'm wondering why this differs between dhcpv4 and dhcpv6?

File: pkg/dhclient/dhcp4.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
+// DefaultScheme for boot file if there are none in the lease
+var DefaultScheme = "tftp"
```
Comment (rjoleary) [inline pkg/dhclient/dhcp4.go:20]: Changing the variable globally is a bit iffy. Is it reasonably possible to pass it as a parameter?

File: pkg/dhclient/dhcp4.go (hunk 1 above)
Comment (colinmitchell-ggl) [reply pkg/dhclient/dhcp4.go:179]: Not sure why but it seems all dhcpv6 I've encountered pass the full url with scheme. It seems it just wasn't needed

File: pkg/dhclient/dhcp4.go (hunk 2 above)
Comment (colinmitchell-ggl) [reply pkg/dhclient/dhcp4.go:20]: That would mean changing the Lease.Boot() interface which and unnecessarily implementing it for v6.
Just for my own education, why would the global be frowned on here? Did you want setters/getters?

**Review by rjoleary — APPROVED** [review]

**Pull request merged**